src/ray/common/BUILD

```diff
@@ -0,0 +1,255 @@
+load("//bazel:ray.bzl", "cc_library")
+
+cc_library(
+    name = "constants",
+    hdrs = ["constants.h"],
+)
+
+cc_library(
+    name = "test_util",
+    hdrs = ["test_util.h"],
+    srcs = ["test_util.cc"],
+    deps = [
+        ":id",
+        ":ray_object",
+        ":network",
+        "//src/ray/util:util",
+        "@com_google_googletest//:gtest",
+        "//src/ray/protobuf:common_cc_proto",
+        "@boost//:optional",
+    ],
+)
+
+cc_library(
+    name = "ray_object",
+    hdrs = [
+        "buffer.h",
+        "ray_object.h"],
+    srcs = ["ray_object.cc"],
+    deps = [
+        "//src/ray/util:util",
+        "//src/ray/protobuf:gcs_cc_proto",
+        ":status",
+        ":id",
+        "@com_google_absl//absl/time",
+        "@com_google_absl//absl/types:optional",
+        "//:aligned_alloc",
+    ]
+)
+
+cc_library(
+    name = "grpc_util",
+    hdrs = ["grpc_util.h"],
+    deps = [
+        "@com_github_grpc_grpc//:grpc++",
+        ":status",
+        ":ray_config",
+        "@com_google_absl//absl/container:flat_hash_map",
+    ]
+)
+
+
+cc_library(
+    name = "memory_monitor",
+    srcs = [
+        "file_system_monitor.cc",
+        "memory_monitor.cc",
+    ],
+    hdrs = [
+        "file_system_monitor.h",
+        "memory_monitor.h",
+    ],
+    deps = [
+        ":asio",
+        "//src/ray/util:util",
+        ":ray_config",
+        "@com_google_absl//absl/strings",
+        "@nlohmann_json",
+        "@com_google_googletest//:gtest_prod",
+    ]
+)
+
+cc_library(
+    name = "runtime_env",
+    srcs = [
+        "runtime_env_common.cc",
+        "runtime_env_manager.cc",
+    ],
+    hdrs = [
+        "runtime_env_common.h",
+        "runtime_env_manager.h",
+    ],
+    deps = [
+        ":id",
+        "//src/ray/protobuf:common_cc_proto",
+        "@com_google_absl//absl/container:flat_hash_map",
+    ]
+)
+
+cc_library(
+    name = "network",
+    srcs = [
+        "network_util.cc",
+        "client_connection.cc"],
+    hdrs = [
+        "client_connection.h",
+        "network_util.h",
+    ],
+    deps = [
+        ":asio",
+        ":id",
+        ":status",
+        "//:node_manager_fbs",
+    ]
+)
+
+cc_library(
+    name = "id",
+    srcs = [
+        "id.cc",
+        "common_protocol.cc",
+    ],
+    hdrs = [
+        "id.h",
+        "common_protocol.h",
+        "id_def.h",
+    ],
+    deps = [
+        "//src/ray/util:util",
+        ":constants",
+        "//src/ray/protobuf:common_cc_proto",
+        "@com_github_google_flatbuffers//:flatbuffers",
+        ":status",
+        "@msgpack",
+    ]
+)
+
+cc_library(
```

Comment:
I actually have no idea if the splitting makes sense. not familiar with the code base at all.